coderd/coderd.go

```diff
@@ -367,6 +369,14 @@ func New(options *Options) *API {
 		DisableSessionExpiryRefresh: options.DeploymentValues.DisableSessionExpiryRefresh.Value(),
 		Optional:                    false,
 	})
+	// Same as the first but it's optional.
+	apiKeyMiddlewareOptional := httpmw.ExtractAPIKeyMW(httpmw.ExtractAPIKeyConfig{
```

Comment:
Would it be cleaner if we allowed `Optional` as a separate argument to this function? Then they could share configs.

Reply:
There are a few values that change between the 3 instances of ExtractAPIKeyMW we have like Optional, RedirectToLogin, etc. Do you propose we add all of these as bool params? Seems janky IMO